circuits/utils/circuitFunctionGetter.ts

```diff
@@ -3,13 +3,27 @@ import {
   circuitFunctionsArray,
 } from "../types/circuitFunctions.types";
 
-export function getCircuitFunctionIndex(funcName: CircuitFunctions) {
+export function getCircuitFunctionIndexByName(funcName: CircuitFunctions) {
   const index = circuitFunctionsArray.indexOf(funcName);
   if (index === -1)
     throw new Error(`function with name ${funcName} does not exist`);
   return index;
 }
 
+export function getCircuitFunctionIndex<
+  T extends CircuitFunctions | CircuitFunctions[],
+>(funcName: T): T extends CircuitFunctions[] ? number[] : number {
+  if (Array.isArray(funcName)) {
+    return funcName.map((name) =>
+      getCircuitFunctionIndexByName(name)
+    ) as T extends CircuitFunctions[] ? number[] : number;
+  } else {
+    return getCircuitFunctionIndexByName(
+      funcName
+    ) as T extends CircuitFunctions[] ? number[] : number;
+  }
+}
+
```

Comment:
Do you need to use generics here to get the return type? 

Simple version
```typescript
export function getCircuitFunctionIndex(
	funcName: CircuitFunctions | CircuitFunctions[]
): number | number[] {
	if (Array.isArray(funcName) {
		return funcName.map((name) => getCircuitFunctionIndexByName(name));
	} else {
		return getCircuitFunctionIndexByName(funcName);
	}
}
```

Reply:
I think we do. It is safer. Return typed depends on the argument type.